pkg/metrics/stats_reporter.go

```diff
@@ -149,3 +222,33 @@ func ReportSystemError(ctx context.Context, errorString string) {
 		systemErrorCount.Add(ctx, 1, attribute.KeyValue{Key: "error", Value: attribute.StringValue(errorString)})
 	}
 }
+
+func ReportRegistryRequestCount(ctx context.Context, statusCode int, registryHost string) {
+	if registryRequestCount != nil {
+		registryRequestCount.Add(ctx, 1, attribute.KeyValue{Key: "status_code", Value: attribute.IntValue(statusCode)}, attribute.KeyValue{Key: "registry_host", Value: attribute.StringValue(registryHost)})
+	}
+}
+
+func ReportAADExchangeDuration(ctx context.Context, duration int64, resourceType string) {
+	if aadExchangeDuration != nil {
+		aadExchangeDuration.Record(ctx, duration, attribute.KeyValue{Key: "resource_type", Value: attribute.StringValue(resourceType)})
+	}
+}
+
+func ReportACRExchangeDuration(ctx context.Context, duration int64, repository string) {
+	if acrExchangeDuration != nil {
+		acrExchangeDuration.Record(ctx, duration, attribute.KeyValue{Key: "repository", Value: attribute.StringValue(repository)})
+	}
+}
+
+func ReportAKVCertificateDuration(ctx context.Context, duration int64, certificateName string) {
+	if akvCertificateDuration != nil {
+		akvCertificateDuration.Record(ctx, duration, attribute.KeyValue{Key: "certificate_name", Value: attribute.StringValue(certificateName)})
+	}
+}
+
+func ReportBlobCacheCount(ctx context.Context, hit bool) {
+	if cacheBlobCount != nil {
```

Comment:
wonder if we should log the err?

Reply:
Sorry I don't think I'm following. Which `err` are you referring to?

Reply:
oops, I mean if we should log error if it's nil.